token-lending/program/src/instruction.rs

```diff
@@ -31,257 +31,264 @@ pub enum LendingInstruction {
+    ///   14 `[]` Token program id
+    ///   15 `[optional]` Reserve liquidity aggregator account
+    ///                     Not required for quote currency reserves.
+    ///                     Must match quote and base currency.
     InitReserve {
         /// Initial amount of liquidity to deposit into the new reserve
         liquidity_amount: u64,
         /// Reserve configuration values
         config: ReserveConfig,
     },
 
-    // 2
-    /// Initializes a new loan obligation.
+    // 3
+    /// Accrue interest and update median quote token price on a reserve.
     ///
     /// Accounts expected by this instruction:
     ///
-    ///   0. `[]` Deposit reserve account.
-    ///   1. `[]` Borrow reserve account.
-    ///   2. `[writable]` Obligation
-    ///   3. `[writable]` Obligation token mint
-    ///   4. `[writable]` Obligation token output
-    ///   5. `[]` Obligation token owner
-    ///   6. `[]` Lending market account.
-    ///   7. `[]` Derived lending market authority.
-    ///   8. `[]` Clock sysvar
-    ///   9. `[]` Rent sysvar
-    ///   10 '[]` Token program id
-    InitObligation,
+    ///   0. `[writable]` Reserve account
+    ///   1. `[]` Clock sysvar
+    ///   2. `[optional]` Reserve liquidity aggregator account
+    RefreshReserve,
 
-    // 3
-    /// Deposit liquidity into a reserve. The output is a collateral token representing ownership
+    // 4
+    /// Deposit liquidity into a reserve in exchange for collateral. Collateral represents a share
     /// of the reserve liquidity pool.
     ///
     /// Accounts expected by this instruction:
     ///
-    ///   0. `[writable]` Source liquidity token account. $authority can transfer $liquidity_amount
-    ///   1. `[writable]` Destination collateral token account.
-    ///   2. `[writable]` Reserve account.
-    ///   3. `[writable]` Reserve liquidity supply SPL Token account.
-    ///   4. `[writable]` Reserve collateral SPL Token mint.
-    ///   5. `[]` Lending market account.
-    ///   6. `[]` Derived lending market authority.
-    ///   7. `[signer]` User transfer authority ($authority).
+    ///   0. `[writable]` Source liquidity token account
+    ///                     $authority can transfer $liquidity_amount.
+    ///   1. `[writable]` Destination collateral token account
+    ///   2. `[writable]` Reserve account
+    ///   3. `[writable]` Reserve liquidity supply SPL Token account
+    ///   4. `[writable]` Reserve collateral SPL Token mint
+    ///   5. `[]` Lending market account
+    ///   6. `[]` Derived lending market authority
+    ///   7. `[signer]` User transfer authority ($authority)
     ///   8. `[]` Clock sysvar
-    ///   9. '[]` Token program id
+    ///   9. `[]` Token program id
     DepositReserveLiquidity {
-        /// Amount to deposit into the reserve
+        /// Amount of liquidity to deposit in exchange for collateral
```

Comment:
Yeah, "collateral" in general is a bit of a confusing abstraction, because a user may think of their reserve deposits as "the collateral" for their loan. I'll rework this and the use of "liquidity" also.

Reply:
Use of deposits/collateral and borrows/liquidity have been differentiated throughout the code. I want to kick this particular proposal to a future PR if that's okay.